### PR TITLE
mp.graphql.showErrorMessage - wildcard

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://github.com/smallrye/smallrye-graphql/issues/519

I have written the test cases but that's not working correctly as Runner is not picking config properties.

### CODE REVIEW

File: server/tck/src/test/resources/META-INF/microprofile-config.properties [hunk 1]
```diff
@@ -2,6 +2,7 @@ smallrye.graphql.printDataFetcherException=true
 smallrye.graphql.tracing.enabled=true
 smallrye.graphql.allowGet=true
 smallrye.graphql.logPayload=true
-mp.graphql.showErrorMessage=java.security.AccessControlException
-mp.graphql.hideErrorMessage=java.io.IOException
+
+mp.graphql.showErrorMessage=java.security.AccessControlException,io.smallrye.graphql.test.apps.exceptionlist.MyUncheckedException
```
Comment (phillip-kruger) [inline server/tck/src/test/resources/META-INF/microprofile-config.properties:6]: `mp.graphql.showErrorMessage=java.security.AccessControlException,io.smallrye.graphql.test.apps.exceptionlist.*`

File: server/tck/src/test/resources/META-INF/microprofile-config.properties [hunk 2]
```diff
@@ -2,6 +2,7 @@ smallrye.graphql.printDataFetcherException=true
...
+mp.graphql.hideErrorMessage=java.io.IOException,org.eclipse.microprofile.graphql.tck.apps.superhero.api.*,io.smallrye.graphql.test.apps.exceptionlist.MyCheckedException
```
Comment (phillip-kruger) [inline server/tck/src/test/resources/META-INF/microprofile-config.properties:7]: `mp.graphql.hideErrorMessage=java.io.IOException,io.smallrye.graphql.test.apps.exceptionlist.*`

**Review by phillip-kruger — CHANGES_REQUESTED** [review]
Change those two config value, then we test both hide and show for wildcard

File: server/implementation/src/main/java/io/smallrye/graphql/spi/config/Config.java [hunk 3]
```diff
@@ -86,8 +86,13 @@ default boolean isListed(Class throwableClass, List<String> classNames) {
...
 
         // Check that specific class
-        if (classNames.contains(throwableClass.getName())) {
-            return true;
+        for (String configuredValue : classNames) {
+            if (configuredValue.equals(throwableClass.getName())) {
+                return true;
+            } else if (configuredValue.endsWith("*")) {
+                String values = configuredValue.substring(0, configuredValue.length() - 2);
+                return throwableClass.getName().startsWith(values);
```
Comment (phillip-kruger) [inline server/implementation/src/main/java/io/smallrye/graphql/spi/config/Config.java:94]: Here only return if true. If false, we still want to iterate the rest of the for loop

Comment (parsharma) [reply server/implementation/src/main/java/io/smallrye/graphql/spi/config/Config.java]: Thanks, you are right. We need to iterate the loop if it is false.

**Comment by phillip-kruger** [comment]
@parsharma - seems like the TCK is passing now ?

**Comment by phillip-kruger** [comment]
Ok, but there are not test cases to test the wildcard ?

**Comment by parsharma** [comment]
Yes, tck is passing now. I will add the test cases, I am just checking why the runner is not picking values.

**Review by phillip-kruger — APPROVED** [review]

**Pull request merged**